bpaotu/bpaotu/views.py

```diff
@@ -555,6 +560,44 @@ def yield_csv_function():
     return response
 
 
+@csrf_exempt
```

Comment:
you've violated the security boundaries we drew on that piece of paper here and in the JS above.

The user-agent (browser) is enforcing the security. The code the user-agent runs can be edited by anyone.

You need to be sending the auth token through to all endpoints, and then those endpoints need to check it and make a security decision. Here you've got the user-agent checking if the token is secure, but all the other endpoints totally open. To bypass this all you would need to do is comment out the check in the client-side JS.